code/src/main/kotlin/com/expediagroup/sdk/core/logging/masking/MaskingPatternBuilder.kt

```diff
@@ -0,0 +1,79 @@
+package com.expediagroup.sdk.core.logging.masking
+
+import com.ebay.ejmask.api.MaskingPattern
+
+/**
+ * Builder class for creating masking patterns.
+ */
+internal class MaskingPatternBuilder {
+    private var globalFields: Set<String> = setOf()
+    private var pathFields: Set<List<String>> = setOf()
+
+    /**
+     * Adds global fields to be masked.
+     *
+     * @param name Vararg of field names to be masked globally.
+     * @return The current instance of MaskingPatternBuilder.
+     */
+    fun globalFields(vararg name: String): MaskingPatternBuilder = apply {
+        globalFields += name.toSortedSet()
+    }
+
+    /**
+     * Adds path-specific fields to be masked.
+     *
+     * @param paths Vararg of lists of field names to be masked by path.
+     * @return The current instance of MaskingPatternBuilder.
+     */
+    fun pathFields(vararg paths: List<String>) = apply {
+        pathFields += paths.map { it.takeLast(2) }
```

Comment:
How will this affect us? Didn't you mention you have a workaround for it?

And to make sure I understand, I wouldn't be able to mask x in `a.b.c.x` without masking `a.x.y.z` because of this limit. Right? Please add a clear example to the doc.